supervisor/container.go

```diff
@@ -252,3 +263,130 @@ func (c *Container) reap() {
 	os.RemoveAll(containerSharedDir)
 	os.RemoveAll(filepath.Join(c.ownerPod.sv.StateDir, c.Id))
 }
+
+func mountToRootfs(m *specs.Mount, rootfs, mountLabel string) error {
+	// TODO: we don't use mountLabel here because it looks like mountLabel is
+	// only significant when SELinux is enabled.
+	var (
+		dest = m.Destination
+	)
+	if !strings.HasPrefix(dest, rootfs) {
+		dest = filepath.Join(rootfs, dest)
+	}
+
+	switch m.Type {
+	case "proc", "sysfs", "mqueue":
+		if err := os.MkdirAll(dest, 0755); err != nil {
+			return err
+		}
+		// Selinux kernels do not support labeling of /proc or /sys
+		return mount.Mount(m.Source, dest, m.Type, strings.Join(m.Options, ","))
+	case "tmpfs":
+		stat, err := os.Stat(dest)
+		if err != nil {
+			if err := os.MkdirAll(dest, 0755); err != nil {
+				return err
+			}
+		}
+		if err := mount.Mount(m.Source, dest, m.Type, strings.Join(m.Options, ",")); err != nil {
+			return err
+		}
+		if stat != nil {
+			if err = os.Chmod(dest, stat.Mode()); err != nil {
+				return err
+			}
+		}
+		return nil
+	case "bind":
+		stat, err := os.Stat(m.Source)
+		if err != nil {
+			// error out if the source of a bind mount does not exist as we will be
+			// unable to bind anything to it.
+			return err
+		}
+		// ensure that the destination of the bind mount is resolved of symlinks at mount time because
+		// any previous mounts can invalidate the next mount's destination.
+		// this can happen when a user specifies mounts within other mounts to cause breakouts or other
+		// evil stuff to try to escape the container's rootfs.
+		if dest, err = symlink.FollowSymlinkInScope(filepath.Join(rootfs, m.Destination), rootfs); err != nil {
+			return err
+		}
+		if err := checkMountDestination(rootfs, dest); err != nil {
+			return err
+		}
+		// update the mount with the correct dest after symlinks are resolved.
+		m.Destination = dest
+		if err := createIfNotExists(dest, stat.IsDir()); err != nil {
+			return err
+		}
+		if err := mount.Mount(m.Source, dest, m.Type, strings.Join(m.Options, ",")); err != nil {
+			return err
+		}
+	case "cgroup":
+		// TODO: We don't support Cgroup currently
+	default:
+		if err := os.MkdirAll(dest, 0755); err != nil {
+			return err
+		}
+		return mount.Mount(m.Source, dest, m.Type, strings.Join(m.Options, ","))
+	}
+	return nil
+}
+
+// checkMountDestination checks to ensure that the mount destination is not over the top of /proc.
+// dest is required to be an abs path and have any symlinks resolved before calling this function.
+func checkMountDestination(rootfs, dest string) error {
+	invalidDestinations := []string{
+		"/proc",
+	}
+	// White list, it should be sub directories of invalid destinations
+	validDestinations := []string{
+		// These entries can be bind mounted by files emulated by fuse,
+		// so commands like top, free displays stats in container.
```

Comment:
It looks like that mount points in white list can be used by `fuse` for better isolation. I'm not sure how can runv make use of this, but maybe we can leave it here. 
It wont harm anyone, right?  If not, I can move mount points in white list(validDestinations ) into black list(invalidDestinations)